lib/galaxy/model/migrations/alembic/versions_gxy/b855b714e8b8_make_api_keys_deleted_non_nullable.py

```diff
@@ -0,0 +1,39 @@
+"""make api_keys deleted non nullable
+
+Revision ID: b855b714e8b8
+Revises: 3356bc2ecfc4
+Create Date: 2023-04-19 18:41:13.500332
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+from galaxy.model.migrations.util import (
+    alter_column,
+    transaction,
+)
+
+# revision identifiers, used by Alembic.
+revision = "b855b714e8b8"
+down_revision = "3356bc2ecfc4"
+branch_labels = None
+depends_on = None
+
+
+table_name = "api_keys"
+column_name = "deleted"
+
+
+def upgrade():
+    with transaction():
+        # Update any existing rows with a deleted=NULL value to be deleted=True.
+        # This will expire any existing API keys that didn't have a deleted value set.
+        table = sa.sql.table(table_name, sa.Column(column_name, sa.Boolean(), nullable=True))
+        op.execute(table.update().where(table.c.deleted.is_(None)).values(deleted=True))
+
+        # Make the column non-nullable.
+        alter_column(table_name, column_name, existing_type=sa.Boolean(), nullable=False, server_default=sa.false())
+
+
+def downgrade():
+    alter_column(table_name, column_name, existing_type=sa.Boolean(), nullable=True, server_default=None)
```

Comment:
```suggestion
    alter_column(table_name, column_name, existing_type=sa.Boolean(), nullable=True, existing_server_default=False)
```

Reply:
This will modify the original column's behavior after a downgrade, right?
With this proposed change, after the `downgrade` any new "deleted=null" values will be defaulted to "deleted=False". I don't think this is what we want unless I'm missing something :thinking:

Reply:
`False` is the current and future default for the `deleted` column.

Reply:
I was mistaken here, we set a `default` but not a `server_default` for this column (actually for any column). Thanks to David, I discovered `server_default` is the one that should be used if you want the default to apply also outside of SQLAlchemy (i.e. when using SQL statements).

Reply:
More on this :point_up: Essentially, even though the resulted value in the database will be the same, `default` is not the same as `server_default` (I'm looking at [these docs](https://docs.sqlalchemy.org/en/20/core/defaults.html)). Since we do not use `server_default` in the model's definition (and why would we), I think it would make sense to ensure the column DDL in the database does not change as a result of running an upgrade->downgrade sequence?